end2end-test-examples/gcs/src/main/java/io/grpc/gcs/JavaApiServicesClient.java

```diff
@@ -0,0 +1,175 @@
+            Runnable task = () -> makeMediaRequest(results, finalI + 1);
+            threadPoolExecutor.execute(task);
+          }
+          break;
+        case METHOD_RANDOM:
+          for (int i = 0; i < args.threads; i++) {
+            int finalI = i;
+            Runnable task =
+                () -> {
+                  try {
+                    makeRandomMediaRequest(results, finalI + 1);
+                  } catch (IOException e) {
+                    e.printStackTrace();
+                  }
+                };
+            threadPoolExecutor.execute(task);
+          }
+          break;
+        case METHOD_WRITE:
+          for (int i = 0; i < args.threads; i++) {
+            int finalI = i;
+            Runnable task = () -> makeInsertRequest(results, finalI + 1);
+            threadPoolExecutor.execute(task);
+          }
+          break;
+        default:
+          logger.warning("Please provide valid methods with --method");
+      }
+      threadPoolExecutor.shutdown();
+      if (!threadPoolExecutor.awaitTermination(30, TimeUnit.MINUTES)) {
+        threadPoolExecutor.shutdownNow();
+      }
+    }
+  }
+
+  public void makeMediaRequest(ResultTable results, int threadId) {
+    for (int i = 0; i < args.calls; i++) {
+      String object = objectResolver.Resolve(threadId, i);
+      long start = System.currentTimeMillis();
+      byte[] content = new byte[0];
+      try {
+        Storage.Objects.Get req = client
+                .objects()
+                .get(args.bkt, object);
+        req.setReturnRawInputStream(true);
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        req.executeMedia().download(output);
+        content = output.toByteArray();
+      } catch (IOException ioException) {
+        System.err.println(ioException);
+      }
+      long dur = System.currentTimeMillis() - start;
+      results.reportResult(args.bkt, object, content.length, dur);
+    }
+  }
+
+  public void makeRandomMediaRequest(ResultTable results, int threadId) throws IOException {
+    Random r = new Random();
+    String object = objectResolver.Resolve(threadId, /*objectId=*/ 0);
+    for (int i = 0; i < args.calls; i++) {
+      long offset = (long) r.nextInt(args.size - args.buffSize) * 1024;
+      long start = System.currentTimeMillis();
+      ByteBuffer buff = ByteBuffer.allocate(args.buffSize * 1024);
+      try {
+        Storage.Objects.Get req = client.objects().get(args.bkt, object);
+        if(offset > 0) {
+          req.getRequestHeaders().setRange(String.format("bytes=%d-", offset));
+        }
+        req.setReturnRawInputStream(false);
+        MediaHttpDownloader mediaHttpDownloader = req.getMediaHttpDownloader();
+        mediaHttpDownloader.setDirectDownloadEnabled(true);
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        req.executeMedia().download(byteArrayOutputStream);
+        buff.put(byteArrayOutputStream.toByteArray());
```

Comment:
We don't need `buff` and instead can do all the post checks relative to the value returned from `byteArrayOutputStream.toByteArray()`

Reply:
Thanks for the review, addressed!